src/trampoline.rs

```diff
@@ -0,0 +1,86 @@
+use neon_runtime::raw::Env;
+use neon_runtime::tsfn::{CallError, ThreadsafeFunction};
+
+use crate::context::TaskContext;
+use crate::result::NeonResult;
+
+pub(crate) type Callback = Box<dyn FnOnce(Env) + Send + 'static>;
+
+pub(crate) struct ThreadsafeTrampoline {
+    tsfn: ThreadsafeFunction<Callback>,
+    ref_count: u32,
```

Comment:
Nitpick: if this were AtomicU32, you could avoid the lock. (You'd be able to use Relaxed operations safely because you're only checking the refcount, not relying on any other data to have been synchronized.)

Reply:
Sadly, the lock is needed because `.reference()`/`.unref()` calls need mutable access to the `ThreadsafeFunction`.

Reply:
Since they require `&mut Context`, it implies the operations are serialized. We could get away with a `Cell` instead.

But, that might be an optimization better left for when it's demonstrated to be a performance bottleneck.

Reply:
Yeah, I don't imagine this being a bottleneck, and it is way too simple to do it with a mutex. I'm open to changing it, though.